unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+    EXPECT_EQ(lp_iviShell->bkgnd_color, ControllerTests::ms_bkgndColor);
+    EXPECT_EQ(lp_iviShell->enable_cursor, ControllerTests::ms_enableCursor);
+    EXPECT_EQ(lp_iviShell->screen_ids.size, sizeof(struct screen_id_info));
+    EXPECT_EQ(custom_wl_list_length(&lp_iviShell->list_surface), 0);
+    EXPECT_EQ(custom_wl_list_length(&lp_iviShell->list_layer), 0);
+    EXPECT_EQ(custom_wl_list_length(&lp_iviShell->list_screen), 0);
+    EXPECT_EQ(custom_wl_list_length(&lp_iviShell->list_controller), 0);
+    EXPECT_NE(lp_iviShell->layer_created.notify, nullptr);
+    EXPECT_NE(lp_iviShell->layer_removed.notify, nullptr);
+    EXPECT_NE(lp_iviShell->surface_created.notify, nullptr);
+    EXPECT_NE(lp_iviShell->surface_removed.notify, nullptr);
+    EXPECT_NE(lp_iviShell->surface_configured.notify, nullptr);
+    EXPECT_NE(lp_iviShell->output_created.notify, nullptr);
+    EXPECT_NE(lp_iviShell->output_destroyed.notify, nullptr);
+    EXPECT_NE(lp_iviShell->output_resized.notify, nullptr);
+    EXPECT_EQ(custom_wl_list_length(&m_westonCompositor.output_created_signal.listener_list), 1);
+    EXPECT_EQ(custom_wl_list_length(&m_westonCompositor.output_destroyed_signal.listener_list), 1);
+    EXPECT_EQ(custom_wl_list_length(&m_westonCompositor.output_resized_signal.listener_list), 1);
+
+    struct screen_id_info *screen_info = NULL;
+    wl_array_for_each(screen_info, &lp_iviShell->screen_ids) {
+        free(screen_info->screen_name);
+    }
+    wl_array_release(&lp_iviShell->screen_ids);
+    free(lp_iviShell->ivi_client_name);
+    free(lp_iviShell->debug_scopes);
+    free(lp_iviShell);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_destroy_success
+ * @brief               Test case of controller_screen_destroy() where valid input params
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_destroy()
+ *                      -# Verification point:
+ *                         +# wl_resource_destroy() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_destroy_success)
+{
+    controller_screen_destroy(nullptr, nullptr);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_clear_cannotGetUserData
+ * @brief               Test case of controller_screen_clear() where wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_clear()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_clear_cannotGetUserData)
+{
+    controller_screen_clear(nullptr, nullptr);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
```

Comment:
should also check the input argument of wl_resource_post_event, for make sure it sends IVI_WM_SCREEN_ERROR_NO_SCREEN